engines/private/private.cpp

```diff
@@ -737,6 +750,158 @@ void PrivateEngine::selectMask(Common::Point mousePos) {
 	}
 }
 
+bool PrivateEngine::selectLocation(Common::Point mousePos) {
+	if (_locationMasks.size() == 0) {
+		return false;
+	}
+	
+	uint i = 0;
+	uint totalLocations = 0;
+	for (NameList::const_iterator it = maps.locationList.begin(); it != maps.locationList.end(); ++it) {
+		const Private::Symbol *sym = maps.locations.getVal(*it);
+		if (sym->u.val) {
+			if (inMask(_locationMasks[i].surf, mousePos)) {
+				for (uint j = 0; j < _diaryPages.size(); j++) {
+					if (_diaryPages[j].locationID == totalLocations + 1) {
+						_currentDiaryPage = j;
+						break;
+					}
+				}
+	
+				_numberClicks++;
+				_nextSetting = _locationMasks[i].nextSetting;
+
+				return true;
+			}
+			i++;
+		}
+		totalLocations++;
+	}
+
+	return false;
+}
+
+bool PrivateEngine::selectDiaryNextPage(Common::Point mousePos) {
+	mousePos = mousePos - _origin;
+	if (mousePos.x < 0 || mousePos.y < 0)
+		return false;
+
+	if (_diaryNextPageExit.rect.contains(mousePos)) {
+		_currentDiaryPage++;
+		_nextSetting = _diaryNextPageExit.nextSetting;
+
+		return true;
+	}
+
+	return false;
+}
+
+bool PrivateEngine::selectDiaryPrevPage(Common::Point mousePos) {
+	mousePos = mousePos - _origin;
+	if (mousePos.x < 0 || mousePos.y < 0)
+		return false;
+
+	if (_diaryPrevPageExit.rect.contains(mousePos)) {
+		_currentDiaryPage--;
+		_nextSetting = _diaryPrevPageExit.nextSetting;
+
+		return true;
+	}
+
+	return false;
+}
+
+bool PrivateEngine::selectMemory(Common::Point mousePos) {
+	if (_memoryMasks.size() == 0) { 
```

Comment:
I think, in this method this check is redundant, since if size is zero, the for() loop will not be executed.